src/core/src/Environment.C

```diff
@@ -219,8 +224,13 @@ BaseEnvironment::BaseEnvironment(
   // If the user passed in an options object pointer, we really shouldn't let
   // ScopedPtr delete their object, so we make a copy.  That way, the dtor
   // will kill this local copy and leave the user's object in tact.
-  if (alternativeOptionsValues != NULL) {
-    m_optionsObj.reset(new EnvOptionsValues(*alternativeOptionsValues));
+  if (initialOptionsValues != NULL) {
+    m_optionsObj.reset(new EnvOptionsValues(*initialOptionsValues));
+  }
+  else {
+    // BMA 20170925: Changed this to default construct the options when none
+    // are passed in, so downstream checks for != NULL may behave differently
+    m_optionsObj.reset(new EnvOptionsValues());
```

Comment:
I'm now default constructing EnvOptionsValues if not passed in.  Is that a problem?  Should I remove checks such as:
  if (m_optionsObj == NULL) return ".";
  queso_require_msg(m_optionsObj, "m_optionsObj variable is NULL");

  What about for EmptyEnvironment?  Is it important that it have options = NULL as opposed to just default constructed options?  The only use case is for member BaseTKGroup::m_emptyEnv which doesn't appear to get used anywhere...

Reply:
We're going with "file options override constructor-time C++ options, right?  In that case default-constructing EnvOptionsValues seems like the right thing to do; it gives us a copy of the default options, unless they've been overridden in C++, and either way they're ready for the "input files can override us still further" code path.

Reply:
The ```return ".";``` does seem redundant now, since that's the default value for ```EnvOptionsValues::m_subDisplayFileName```.

Reply:
I'd leave the queso_require in there.  Since it's not in user hands anymore we could turn it into a queso_assert if you like, but it's also not in an inner loop anymore so there's no compelling reason to.

Reply:
And there seems no reason not to give EmptyEnvironment default options, to make it less useless.